lncfg/protocol.go

```diff
@@ -0,0 +1,15 @@
+package lncfg
+
+// ProtocolOptions is a struct that we use to be able to test backwards
+// compatibility of protocol additions, while defaulting to the latest within
+// lnd, or to enable experimental protocol changes.
+type ProtocolOptions struct {
+	// LegacyProtocol is a sub-config that houses all the legacy protocol
+	// options.  These are mostly used for integration tests as most modern
+	// nodes shuld always run with them on by default.
+	LegacyProtocol `group:"legacy" namespace:"legacy"`
+
+	// ExperimentalProtocol is a sub-config that houses any experimental
+	// protocol features that also require a build-tag to activate.
+	ExperimentalProtocol
```

Comment:
do we want to add `group:"experimental" namespace:"experimental"`?

doesn't look like the itests had to change, so it just get embedded? just thinking we might want the isolation in case we want to change the final naming structure when it becomes fully enabled

Reply:
Opted to not make a new namespace so it wouldn't break the itests, nor the deployment of those that may be running with the current flag as is. Yeah as is since we just embed it you don't need to specify anything extra on the command line. Once it's fully enabled, I imagine it would just be the default (like when we've added other new channel types), and this config param would be removed.

Reply:
There is an error in description still, it says "WumboChans should be set if we want to **disable** support for wumbo..." ?

Reply:
Nice catch @vegardengen, fixed! It was opt-out initially, but now we've modified it to be opt-in.